src/editor/editorinteractive.cc

```diff
@@ -951,6 +1287,621 @@ void EditorInteractive::map_changed(const MapWas& action) {
+								for (const auto& pair : ps->ware_queues) {
+									const char* name =
+									   cs.owner().tribe().get_ware_descr(pair.first)->name().c_str();
+									write("%s:set_desired_fill(\"%s\", %u, true)", var, name,
+									      pair.second.desired_fill);
+									write(
+									   "%s:set_priority(\"%s\", %d, true)", var, name, pair.second.priority);
+								}
+								for (const auto& pair : ps->worker_queues) {
+									write("%s:set_desired_fill(\"%s\", %u, true)", var,
+									      cs.owner().tribe().get_worker_descr(pair.first)->name().c_str(),
+									      pair.second.desired_fill);
+								}
+								if (upcast(const Widelands::TrainingsiteSettings, ts, cs.get_settings())) {
+									write("%s.setting_soldier_capacity = %u", var, ts->desired_capacity);
+								}
+							} else if (upcast(const Widelands::WarehouseSettings, ws, cs.get_settings())) {
+								write("%s.setting_launch_expedition = %s", var,
+								      ws->launch_expedition ? "true" : "false");
+								for (const auto& pair : ws->ware_preferences) {
+									write("%s:set_setting_warehouse_policy(\"%s\", \"%s\")", var,
+									      cs.owner().tribe().get_ware_descr(pair.first)->name().c_str(),
+									      stockpolicystring(pair.second));
+								}
+								for (const auto& pair : ws->worker_preferences) {
+									write("%s:set_setting_warehouse_policy(\"%s\", \"%s\")", var,
+									      cs.owner().tribe().get_worker_descr(pair.first)->name().c_str(),
+									      stockpolicystring(pair.second));
+								}
+							}
+						} break;
+						case Widelands::MapObjectType::DISMANTLESITE: {
+							const Widelands::DismantleSite& ds =
+							   dynamic_cast<const Widelands::DismantleSite&>(*f.get_immovable());
+							const int16_t xpos = ds.get_position().x;
+							const int16_t ypos = ds.get_position().y;
+							write("%s = %s:place_building(\"%s\", "
+							      "%s:get_field(%d, %d), false, true)",
+							      var, pvar[ds.owner().player_number() - 1].c_str(),
+							      ds.building().name().c_str(), autogen_map_name.c_str(), xpos, ypos);
+							write("%s:dismantle()", var);
+							write("%s:get_field(%d, %d).immovable.has_builder = %s",
+							      autogen_map_name.c_str(), xpos, ypos,
+							      ds.has_builder(egbase()) ? "true" : "false");
+						} break;
+						default:
+							break;
+						}
+					}
+				}
+			}
+		}
+
+		for (const Widelands::Economy* e : economies_to_save) {
+			const Widelands::Coords flag = e->get_arbitrary_flag()->get_position();
+			if (e->type() == Widelands::wwWARE) {
+				write("%s = %s:get_field(%u, %u).immovable.ware_economy", var, autogen_map_name.c_str(),
+				      flag.x, flag.y);
+				for (Widelands::DescriptionIndex di : e->owner().tribe().wares()) {
+					const Widelands::WareDescr& d = *egbase().tribes().get_ware_descr(di);
+					if (d.has_demand_check(e->owner().tribe().name()))
+						write("%s:set_target_quantity(\"%s\", %u)", var, d.name().c_str(),
```

Comment:
We should only write these if they differ from the default target quantity.

Reply:
What if the scenario designer wants to set each target quantity individually and some of his preferred values happen to be the default target? Then the configuration will be messed up if we ever change the defaults.

Reply:
I didn't think of that. There is no perfect solution then.